### PR TITLE
Add AWS event-normalizer

### LINKED ISSUE #647: New Middleware Proposal: SNS JSON Parser
Proposal for a new parsing middleware that could transform inputs from sns.

https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html

See `sqs-json-body-parser` for what it could look like: https://github.com/middyjs/middy/tree/main/packages/sqs-json-body-parser

Please 👍🏻 this issue if it would have value to you. I'll build it if there > 10 👍🏻

If you can comment with your use case below, that would be helpful.

### LINKED ISSUE #648: New Middleware Proposal: Kinesis JSON Parser
Proposal for a new parsing middleware that could transform inputs from kinesis.

- [ ] Firehose https://docs.aws.amazon.com/lambda/latest/dg/services-kinesisfirehose.html
  - [ ] can be nested in sns
- [ ] streams https://docs.aws.amazon.com/lambda/latest/dg/with-kinesis.html

See `sqs-json-body-parser` for what it could look like: https://github.com/middyjs/middy/tree/main/packages/sqs-json-body-parser

Please 👍🏻 this issue if it would have value to you. I'll build it if there > 10 👍🏻

If you can comment with your use case below, that would be helpful.

### PR DESCRIPTION
**Pull request by willfarrell**
Closes #647 #648 #652

### CODE REVIEW

**Comment by willfarrell** [comment]
@lmammino Been thinking about on this one. What if there was one middleware that did json parsing of all aws events, including nested chaining (sns -> sqs). Call it `event-json-parser`, then flag `sqs-json-body-parser` as a duplicate? thoughts?

File: packages/sqs-json-body-parser/__tests__/index.js [hunk 1]
```diff
@@ -19,8 +19,8 @@ test.afterEach((t) => {
 test.serial('parses each body payload', async (t) => {
   const request = { event }
   const bodys = [{ one: 1 }, { two: 2 }, { three: 3 }]
-  event.Records.push({ ...event.Records[0] })
-  event.Records.push({ ...event.Records[0] })
+  event.Records.push(JSON.parse(JSON.stringify(event.Records[0])))
+  event.Records.push(JSON.parse(JSON.stringify(event.Records[0])))
```
Comment (lmammino) [inline packages/sqs-json-body-parser/__tests__/index.js:23]: is this to do a deep copy of the objects?

**Review by lmammino — APPROVED** [review]

File: packages/sqs-json-body-parser/__tests__/index.js (hunk 1 above)
Comment (willfarrell) [reply packages/sqs-json-body-parser/__tests__/index.js:23]: yes, it was a regression.

Comment (willfarrell) [reply packages/sqs-json-body-parser/__tests__/index.js:23]: * and to prevent mutations of nested objects

**Pull request merged**